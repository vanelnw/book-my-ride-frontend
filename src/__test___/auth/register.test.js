import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../redux/store';

import RegisterScreen from '../../components/auth/Register';

describe('To test Signup component', () => {
  let signup;
  beforeAll(async () => {
    signup = renderer
      .create(
        <Provider store={store}>
          <Router>
            <RegisterScreen />
          </Router>
        </Provider>,
      )
      .toJSON();
  });

  it('The component should render', () => {
    expect(signup).toBeTruthy();
  });

  it('It should match the related snapshot', () => {
    expect(signup).toMatchSnapshot();
  });

  it('Should have a button with sign up text', () => {
    const btn = screen.findByTestId('submit');
    waitFor(() => expect(btn).toHaveTextContent('Sign up'));
  });

  it('renders all form fields', () => {
    waitFor(() => expect(screen.getByLabelText('Name')).toBeInTheDocument());
    waitFor(() => expect(screen.getByLabelText('Email')).toBeInTheDocument());
    waitFor(() => expect(screen.getByLabelText('Password')).toBeInTheDocument());
    waitFor(() => expect(screen.getByLabelText('Confirm Password')).toBeInTheDocument());
  });
});
