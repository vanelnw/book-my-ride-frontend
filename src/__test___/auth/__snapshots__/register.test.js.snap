// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`To test Signup component It should match the related snapshot 1`] = `
<div
  className="registerPage"
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="top"
    >
      <header>
        Sign Up
      </header>
      <div
        className="input-field reg-field"
      >
        <input
          id="name"
          name="name"
          onChange={[Function]}
          placeholder="Name"
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="input-field reg-field"
      >
        <input
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          required={true}
          type="email"
          value=""
        />
      </div>
      <div
        className="input-field reg-field"
      >
        <input
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
      </div>
      <div
        className="input-field reg-field"
      >
        <input
          id="confirmPassword"
          name="confirmPassword"
          onChange={[Function]}
          placeholder="confirmPassword"
          required={true}
          type="password"
          value=""
        />
      </div>
      <button
        className="submit"
        disabled={false}
        type="submit"
      >
        Sign Up
      </button>
      <div
        className="two-col"
      >
        <div
          className="one"
        >
          <p>
            Already have an account? 
          </p>
          <p>
            <a
              className="whiteLink"
              href="/login"
              onClick={[Function]}
            >
              Login Here
            </a>
          </p>
        </div>
      </div>
    </div>
  </form>
</div>
`;
